rfcs/0132-meson-builds-nix.md

```diff
@@ -0,0 +1,124 @@
+---
+feature: meson_builds_nix
+start-date: 2022-08-25
+author: Anderson Torres
+co-authors: @p01arst0rm
+shepherd-team:
+shepherd-leader:
+related-issues:
+---
+
+# Summary
+[summary]: #summary
+
+Use meson as an alternative build system for the reference implementation of Nix.
+
+# Motivation
+[motivation]: #motivation
+
+Currently, the reference implementation of Nix evaluator and its companion toolset generated from the Nix source tree are built using the typical `./configure` shell script that relies on autoconf and the standard GNU Make utility.
+
+This build system became clunky and plastered, and consequently hard to understand, modify, improve and port to other systems besides Linux.
+
+Such state of things hinders development, specially outside the Linux and NixOS niches.
+
+In light of this, we propose a novel, from-scratch alternative build infrastructure.
+
+We expect to accomplish, among other goals,
+
+- better code structuring;
+- improved cross-platform support, especially in other programming environments, including but not limited to Unix-like operating systems;
+- shorter build times;
+- an overall improved user experience.
```

Comment:
possible additional points:

- better buildsystem documentation
- improved integration of unit testing